appscale/tools/agents/azure_agent.py

```diff
@@ -0,0 +1,694 @@
+
+    keyname = parameters[self.PARAM_KEYNAME]
+    private_key_path = LocalState.LOCAL_APPSCALE_PATH + keyname
+    public_key_path = private_key_path + ".pub"
+
+    with open(public_key_path, 'r') as pub_ssh_key_fd:
+      pub_ssh_key = pub_ssh_key_fd.read()
+
+    key_path = "/home/{}/.ssh/authorized_keys".format(self.ADMIN_USERNAME)
+    public_keys = [SshPublicKey(path=key_path, key_data=pub_ssh_key)]
+    ssh_config = SshConfiguration(public_keys=public_keys)
+    linux_config = LinuxConfiguration(disable_password_authentication=True,
+                                      ssh=ssh_config)
+    os_profile = OSProfile(admin_username=self.ADMIN_USERNAME,
+                           computer_name=vm_network_name,
+                           linux_configuration=linux_config)
+
+    hardware_profile = HardwareProfile(
+      vm_size=VirtualMachineSizeTypes.standard_a3)
+
+    network_profile = NetworkProfile(
+      network_interfaces=[NetworkInterfaceReference(id=network_id)])
+
+    virtual_hd = VirtualHardDisk(
+      uri='https://{0}.blob.core.windows.net/vhds/{1}.vhd'.
+        format(storage_account, vm_network_name))
+
+    image_hd = VirtualHardDisk(uri=parameters[self.PARAM_IMAGE_ID])
+    os_type = OperatingSystemTypes.linux
+    os_disk = OSDisk(os_type=os_type, caching=CachingTypes.read_write,
+                     create_option=DiskCreateOptionTypes.from_image,
+                     name=vm_network_name, vhd=virtual_hd, image=image_hd)
+
+    compute_client.virtual_machines.create_or_update(
+      resource_group, vm_network_name, VirtualMachine(location=zone,
+                                                      os_profile=os_profile,
+                                                      hardware_profile=hardware_profile,
+                                                      network_profile=network_profile,
+                                                      storage_profile=StorageProfile(
+                                                        os_disk=os_disk)))
+
+    # Sleep until an IP address gets associated with the VM.
+    sleep_time = 1
+    while True:
+      public_ip_address = network_client.public_ip_addresses.get(resource_group,
+                                                                 vm_network_name)
+      if public_ip_address.ip_address:
+        print('Azure VM is available at {}'.format(public_ip_address.ip_address))
+        break
+      AppScaleLogger.verbose("Waiting {} second(s) for IP address to be "
+        "available".format(sleep_time), verbose)
+      time.sleep(sleep_time)
+      sleep_time = min(sleep_time * 2, 20)
+
+  def associate_static_ip(self, instance_id, static_ip):
+    """ Associates the given static IP address with the given instance ID.
+
+    Args:
+      instance_id: A str that names the instance that the static IP should be
+        bound to.
+      static_ip: A str naming the static IP to bind to the given instance.
+    """
+
```

Comment:
Right now, we do not enter the block of code which calls this method because static ip is not passed in the AppScalefile. So it works without a pass.